### ISSUE TITLE
Feature Request: Mac - App continues to run in the background

### ISSUE THREAD

**Issue by Lightsout565** [chrisknepper/android-messages-desktop#8]
Mac Requests:

- [ ] Add Icon to Menu Bar; Have menu bar change number or highlight blue when a new message shows up

- [ ] Prevent Cmd+Q from fully quitting the app, allowing you to get notifications for new messages still

- [ ] Cmd+W is broken; doesn't close the window.

**Comment by mwjburton** [comment]
Cmd+H to hide the window would also be very useful. Thanks

**Comment by chrisknepper** [comment]
Hey @Lightsout565 , I have some questions about your requests for macOS:

1. For "Add Icon to Menu Bar," do you mean an icon in the top-right of the system menu bar (as in the attached screenshot)?
2. Most mac apps in my experience will fully quit with the Cmd+Q shortcut. I can definitely make it so the app remains open when the window is closed via the red button, but I want to be clear: you would also like the app to remain open (but have the window disappear) when Cmd+Q is pressed?
3. Would you both be fine with Cmd+W and Cmd+H having the same behavior of closing (hiding) the window, or would you like more specific behavior for each?

**Comment by rjxeon** [comment]
Hey @chrisknepper great app, just switched over to Android from iOS and this was missing having messages on my Mac. 

I agree with @Lightsout565 regards the Application staying open once you have hit the red button, would be a nice feature so it stays in the dock and I can still get Notifications. Everything else seems to run perfect for me. If you need anything new testing give me a shout.

**Comment by chrisknepper** [comment]
@rjxeon Thanks for the kind words! I have the app staying open when the window closes locally, just need to test it some more. I should tentatively release an update enabling this tonight.

**Comment by rjxeon** [comment]
@chrisknepper Great look forward to it!, any Chance of notification bubble counter on the dock icon? Hard to see if you missed a notification

**Comment by Lightsout565** [comment]
@chrisknepper Cmd+W is the common user interaction for closing a window, without quitting the app entirely. Also, regarding your earlier post Cmd+H and Cmd+W are very different commands. Cmd+W closes the window and terminates any instance of the app window, where as Cmd+H hides the app window without destroying it. For example, say I have Spotify open and I'm on an artist page. When I press Cmd+H, it hides the window, but when I click the Spotify app icon in the dock I get the same window, in the same artist page I left it. If Cmd+W, the app is still open but I have terminated and closed the app window. When I press the Spotify icon this time, it will load up a new Spotify window instance and take me to the homepage.

**Comment by kidbrax** [comment]
regarding the cmd-Q to quit i think it should follow best practices and fully quit the app but provide a separate helper if you want to keep something running in the background. in other words, i wouldn't want to quit the app only to find out that hasn't really quit unless i explicitly hose that

**Comment by Lightsout565** [comment]
I agree with @kidbrax. Cmd+Q to quit is best practice and should be the default. Maybe having it as an option in settings as you mention to keep the app running in the background for those of us that want to app open all the time for incoming notifications.

I've actually added Android Messages as a login item for my account (Hide), so it's pretty much always running on my machine.

**Comment by rjxeon** [comment]
@Lightsout565 I think most apps like Messages tend to be CMD + W is quit but keeps running in the background ( light still on in the dock ), and CMD + Q could then be a complete quit out of the app? Notification bubble is working great now though.

**Comment by kidbrax** [comment]
@rjxeon on Mac, cmd-W means close the current window, not sure about other platforms.

**Comment by chrisknepper** [comment]
Alrighty, so it sounds like Cmd+W to close the window but keep running should definitely make it in. Looking at Google Chrome, Cmd+W closes the current window but Chrome is also a multi-window application whereas this one really isn't. In v0.1.0, I added Cmd+H as a shortcut to hide the window without quitting the application since it was requested.

As long as Electron lets me have 2 keyboard shortcuts (which they call "Accelerators") that do the same thing, I'm all for it! Thanks again for your feedback all!

**Comment by chrisknepper** [comment]
Also, would you all prefer that this new shortcut live under a new `File` menu or `Window` menu in the Mac menu bar? I've seen apps do it under both, but I believe `File` is more common. Either way it's a new top-level menu.
<img width="253" alt="screen shot 2018-06-28 at 10 31 38 pm" src="https://user-images.githubusercontent.com/4976096/42070191-40146842-7b23-11e8-8ef7-fda9279e163d.png">

**Comment by Lightsout565** [comment]
Also Cmd M to minimize is also very common. Usually found under a "Window"
option along with File, Help, Etc

**Comment by kidbrax** [comment]
@chrisknepper for cmd-q and cmd-h it should live under the main application menu, "Android Messages" but cmd-w should be under the File menu I believe.  Thanks for adding these in, it's a great app!

**Comment by kidbrax** [comment]
Just downloaded v1 to try the cmd-h to hide.  It doesn't seem to work correctly though.  It hides the window but the app stays in front.  It should hide the whole app and make the next app active.

**Comment by LastInLine** [comment]
@kidbrax is right, cmd-h hides the window but doesn't switch to the next app.  Right now it's working kind of like cmd-m should except without the genie effect to the dock.

@chrisknepper I don't know if you ever got an answer to question 1 in comment 4 but yes, an icon there to show that it's running in the background would be ideal, along with some indicator of an unread message when one comes in.

**Comment by chrisknepper** [comment]
Everything in this should be covered by v0.3.0, with the exception of the "tray" icon highlighting when a notification comes in for macOS. I believe we should open another issue for that because the UX implications of that also apply to Windows and Linux.

**Comment by danieltaylor** [comment]
Just noticed cmd-w didn't make it into the release, but I would love to see it included.  As a refresher, standard practice with Mac apps is for cmd-w to close the current window but leave the app running.
